src/controller/CHIPDeviceController.cpp

```diff
@@ -206,7 +201,10 @@ CHIP_ERROR DeviceController::Init(NodeId localDeviceId, ControllerInitParams par
     VerifyOrReturnError(params.operationalCredentialsDelegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
     mOperationalCredentialsDelegate = params.operationalCredentialsDelegate;
 
-    ReturnErrorOnFailure(LoadLocalCredentials(fabric));
+    if (LoadLocalCredentials() != CHIP_NO_ERROR)
+    {
+        ReturnErrorOnFailure(GenerateLocalCredentials());
```

Comment:
"Local credentials" cannot be generated in general. We need to address the issue that the application should have "self-commissioned" the device before initializing this class. We should discuss as a SDK team the best way to go there.

Reply:
I agree. This is not ideal. The main issue for me is when device attestation is integrated, how to generate a DAC for commissioner. The ideal approach is that the commissioner object is provided with OpCreds, rather than it generating using CA.

Reply:
There is no DAC for the commissioner. The commissioner does not need to attest itself.

I agree that:

> The ideal approach is that the commissioner object is provided with OpCreds, rather than it generating using CA.

Reply:
Resolving commissioner's credentials related comments here for now. We'll move it out of this class (to the application) in a separate PR.

Reply:
Submitted this issue: https://github.com/project-chip/connectedhomeip/issues/8910